### PR TITLE
fix(policies): Fix policy reference

### LINKED ISSUE #1448: bug: wrong policy reference in attestations
Looks like policy reference URIs are broken. This is from a recent attestation:
```
        {
...
            "digest": {
              "sha256": "c1ad64c21bac1bc5df57ace6f4d432eaaf5b8ef1f3ae28b98c6888c3f4e9ad31"
            },
            "name": "chainloop://api.app.chainloop.dev/https://api.app.chainloop.dev/v1/policies/cves-in-kev"
          },
...
        }
```

### PR DESCRIPTION
**Pull request by jiparis**
This PR fixes the way policy references are calculated in the CLI and does some refactor by removing protobuf dependency in this calculation. Once resolved, a descriptor is created containing the URI, Name, OrgName (for custom remote policies) and Digest, which is then used through the attestation process. Previously, a in-toto protobuf message was used instead.

Fixes #1448

### CODE REVIEW

File: app/controlplane/pkg/policies/policyprovider.go [hunk 1]
```diff
@@ -114,75 +116,60 @@ func (p *PolicyProvider) queryProvider(path, digest, orgName, token string, out
...
-		return nil, fmt.Errorf("error parsing policy provider URL: %w", err)
-	}
-
-	if uri.Host == "" {
-		return nil, fmt.Errorf("invalid policy provider URL")
-	}
-
-	if path == "" || digest == "" {
-		return nil, fmt.Errorf("both path and digest are mandatory")
+func createRef(policyURL *url.URL, name, digest, orgName string) *PolicyReference {
```
Comment (migmartri) [inline app/controlplane/pkg/policies/policyprovider.go:166]: I am assuming that this method will be used also for groups?

File: pkg/policies/loader.go [hunk 2]
```diff
@@ -44,20 +43,60 @@ const (
 
 // Loader defines the interface for policy loaders from contract attachments
 type Loader interface {
-	Load(context.Context, *v1.PolicyAttachment) (*v1.Policy, *v12.ResourceDescriptor, error)
+	Load(context.Context, *v1.PolicyAttachment) (*v1.Policy, *PolicyDescriptor, error)
+}
+
+// PolicyDescriptor Represents a policy reference. Used as FQ references.
+type PolicyDescriptor struct {
```
Comment (migmartri) [inline pkg/policies/loader.go:50]: so in the end you moved away from intoto descriptor why?

**Review by migmartri — APPROVED** [review]

File: pkg/policies/loader.go (hunk 2 above)
Comment (jiparis) [reply pkg/policies/loader.go:50]: This is internal. The final representation in the attestation is still an in-toto descriptor, but I didn't want to get locked into any structure in the internal logic for policies. Actually I think we should apply the same to the rest of the policies logic.

Comment (migmartri) [reply pkg/policies/loader.go:50]: oh, I get it, and agree!

**Pull request merged**